shift_detector/Detector.py

```diff
@@ -35,48 +36,47 @@ def __init__(self,
         else:
             raise Exception("df2 is not a dataframe or a string")
 
-        self.checks_to_run = []
         self.check_reports = []
         self.store = Store(self.df1, self.df2)
 
-        logger.info("Used columns: {}".format(' '.join(column_names(self.store.columns))))
+        logger.info("Used columns: {}".format(', '.join(column_names(self.store.columns))))
 
-    def add_checks(self, checks):
+    def run(self, *checks):
         """
-        Add checks to the detector
-
-        :param checks: single or list of Checks
-        """
-        if isinstance(checks, Check):
-            checks_to_run = [checks]
-        elif isinstance(checks, list) and all(isinstance(check, Check) for check in checks):
-            checks_to_run = checks
-        else:
-            raise Exception("All elements in checks should be a Check")
-        self.checks_to_run += checks_to_run
-
-    def run_checks(self) -> List[Report]:
-        """
-        Execute the checks to run.
-
-        :return: list of Reports that resulted from the checks
+        Run the Detector with the checks to run.
+        :param checks: checks to run
         """
-        return [check.run(self.store) for check in self.checks_to_run]
+        if not checks:
+            raise Exception("Please include checks)")
 
-    def run(self):
-        """
-        Run the Detector.
-        """
-        if not self.checks_to_run:
-            raise Exception("Please use the method add_checks to add checks, "
-                            "that should be executed, before calling run()")
+        if not all(isinstance(check, Check) for check in checks):
+            raise Exception("All elements in checks should be a Check")
```

Comment:
include type of objects that aren't of type `Check` in Exception message
use `ValueError` Execption